pkg/webhook/mutation.go

```diff
@@ -162,9 +170,56 @@ func (h *mutationHandler) Handle(ctx context.Context, req admission.Request) adm
 }
 
 func (h *mutationHandler) mutateRequest(ctx context.Context, req admission.Request) (admission.Response, error) {
-	// TODO: place mutation logic here
-	mutatedObject := req.Object
-	resp := admission.PatchResponseFromRaw(req.Object.Raw, mutatedObject.Raw)
+
+	okResp := admission.Allowed("")
+	ns := &corev1.Namespace{}
+
+	// if the object being mutated is a namespace itself, we use it as namespace
+	if req.Kind.Kind == namespaceKind && req.Kind.Group == "" {
+		req.Namespace = ""
+		obj, _, err := deserializer.Decode(req.Object.Raw, nil, &corev1.Namespace{})
+		if err != nil {
+			return admission.Errored(int32(http.StatusInternalServerError), err), nil
+		}
+		ok := false
+		ns, ok = obj.(*corev1.Namespace)
+		if !ok {
+			return admission.Errored(int32(http.StatusInternalServerError), errors.New("failed to cast namespace object")), nil
+		}
+	} else {
+		if err := h.client.Get(ctx, types.NamespacedName{Name: req.AdmissionRequest.Namespace}, ns); err != nil {
+			if !k8serrors.IsNotFound(err) {
+				log.Info("Namespace not found", "name", req.AdmissionRequest.Namespace)
```

Comment:
nit: "error retrieving namespace"

This is because we know that the error is not a "not found" error

Reply:
Done. I also made the logs slightly different, to facilitate understanding which one is happening.

Unrelated: the namespace logic is the same as the policy one. What is the difference between using the client and the apiReader ? I always thought using the client would hit the api server anyway.

Reply:
The standard client hits the object cache (which is populated from a `watch`). By using `reader` we guarantee we are hitting the API server directly.

Basically we fall back to the API server on a cache miss

Reply:
Ah, I thought it was a regular client-go client, turns out is more complex than that.
Thanks.